src/name_core.jl

```diff
@@ -118,8 +118,9 @@ Returns the values of the `named_inds`, sorted as per the order they appear in `
 with any missing dimnames, having there value set to `:`.
 An error is thrown if any dimnames are given in `named_inds` that do not occur in `dimnames`.
 """
-function order_named_inds(dimnames::Tuple{Vararg{Symbol,N}}; named_inds...) where {N}
-    # 0-Allocations: see `@code_typed (()->order_named_inds((:a, :b, :c), (b=1, c=2)))()`
+order_named_inds(dimnames::Tuple; named_inds...) = order_named_inds(dimnames, named_inds.data)
```

Comment:
Can we just get rid of this method entirely,
and demand that the `named_inds` get passed as a NamedTuple 
(or maybe also allow a `Base.Iterators.Pairs`)

Would also need to get rid of the one here:
https://github.com/invenia/NamedDims.jl/blob/f8f77a5119e589eae6f70e548836f7262767b263/src/wrapper_array.jl#L160

and change this call 
https://github.com/invenia/NamedDims.jl/blob/f8f77a5119e589eae6f70e548836f7262767b263/src/wrapper_array.jl#L169
and this call
https://github.com/invenia/NamedDims.jl/blob/f8f77a5119e589eae6f70e548836f7262767b263/src/wrapper_array.jl#L195

Reply:
You could, I tried some such things, and didn't find a way to get to zero allocations & full speed. So this was the minimal way to change things. It's a clear improvement, and if I find a complete solution I'll be sure to make a PR.

Reply:
Thats fair.
I  am happy to merge the PR with this still in it